scripts/package-json-merge.js

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env node
+
+// This script is used to resolve `package.json` merge conflicts and requires
+// the repository to be in a very specific state. See `README.md` and
+// `.github/workflows/update.yml`.
+
+// When a repository using perseids-publications/treebank-template wants to
+// update the template code, the workflow will update the local repository
+// to match the template then run some git commands that check out files like
+// `src/config.json` (see `scripts/setup-update.sh`).
+// The only file that should have conflicts is `package.json`. This script uses
+// the git command line to get the `package.json` from origin/master and reads
+// the local `package.json` file (which matches source/master).
+// It takes the "name", "version", and "homepage" from the origin/master
+// version, keeps the rest the same, then writes the result to `package.json`.
+
+const { execSync } = require('child_process');
+
+const { readFileSync, writeFileSync } = require('fs');
+
+// Read the old package.json with `git show`
+const oldPackage = JSON.parse(
+  execSync('git show origin/master:package.json'),
+);
+
+// Read the new package.json - currently in the repository
+const newPackage = JSON.parse(
+  readFileSync('package.json', { encoding: 'utf8', flag: 'r' }),
+);
+
+// Replace the relevant fields
+[
+  'name',
+  'version',
+  'homepage',
+].forEach((key) => {
+  newPackage[key] = oldPackage[key];
+});
+
+// Write to package.json
+writeFileSync('package.json', `${JSON.stringify(newPackage, null, 2)}\n`);
```

Comment:
That's a good question. The way I was thinking about the versioning of repositories that use the template is that the `version` field would be for the version of the treebank files and not the code. There would also be a [template version](https://github.com/perseids-publications/treebank-template/commit/146f8e6a099c70d1f6223238494f56a3c2ea8e23) in `package.json` that shows version of the template.

Right now creating a new release requires manually updating the `package.json` and then creating the release on GitHub. But maybe we could use a GitHub workflow like this with the [https://github.com/actions/create-release](create-release) action? Then, whenever a user has added enough new treebanks they're ready to create a release, they could use the GitHub Actions interface the same way this action is used.

Doing this might even be a way to get around the [issue](https://github.com/perseids-publications/treebank-template/issues/23#issuecomment-628854252) with the GitHub Zenodo integration not generating a DOI until after the archive is created. The "new release" action could use the [Zenodo API](https://developers.zenodo.org/#representation) to pre-reserve a DOI, update `src/config.json`, commit and push, then create a release on GitHub.

Reply:
that's a great idea.